kebechet/managers/thoth/thoth.py

```diff
@@ -0,0 +1,141 @@
+
+    @property
+    def sha(self):
+        """Get SHA of the current head commit."""
+        return self.repo.head.commit.hexsha
+
+    # TODO: determine how branch name will be constructed
+    def _construct_branch_name(self) -> str:
+        """Construct branch name for the updated dependency."""
+        return f"{_BRANCH_NAME}-{self.sha}"
+
+    
+    def _git_push(self, commit_msg: str, branch_name: str, files: list, force_push: bool = False) -> None:
+        """Perform git push after adding files and giving a commit message."""
+        #  self.repo.git.checkout('HEAD', b=branch_name)
+        self.repo.index.add(files)
+        self.repo.index.commit(commit_msg)
+        self.repo.remote().push(branch_name, force=force_push)
+
+    
+    def _open_merge_request(self, branch_name: str, labels: list, files: list) -> typing.Optional[int]:
+        """Open a pull/merge request for dependency update."""
+        commit_msg = "Auto generated update"
+        body = "Pipfile.lock updated by kebechet-thoth manager"
+
+        # Delete branch if it didn't change Pipfile.lock
+        diff = self.repo.git.diff('master', files)
+        if diff == "":
+            return
+        
+        # push force always to keep branch up2date with the recent master and avoid merge conflicts.
+        self._git_push(":pushpin: " + commit_msg, branch_name, files, force_push=True)
+        
+        for mr in self._cached_merge_requests:
+            if mr.head_branch_name == branch_name:
+                return
+        
+        merge_request = self.sm.open_merge_request(commit_msg, branch_name, body, labels)
+        return merge_request
+
+    @staticmethod
+    def _advise_wrap():
+        with open("Pipfile", "r") as pipfile, open("Pipfile.lock", "a+") as piplock:
+            # TODO: take into account config values for recommendation type,
+            #       limit latest versions, no static analysis, no wait,
+            #       limit, count, debug
+            res = lib.advise(pipfile.read(), piplock.read())
+        return res
+
+    @staticmethod
+    def _write_advise(adv_results: list):
+        lock_info = adv_results[0]["report"][0][1]["requirements_locked"]
+        with open("Pipfile.lock", "w+") as f:
+            f.write(json.dumps(lock_info))
+        return
+
+
+    def run(self, labels: list, rectype: str):
+        with cloned_repo(self.service_url, self.slug, depth=1) as repo:
+            self.repo = repo
+            branch_name = self._construct_branch_name()
+            branch = self.repo.git.checkout('-B', branch_name)
+            self._cached_merge_requests = self.sm.repository.merge_requests
+            if os.path.isfile('Pipfile'):
+                res = self._advise_wrap()
+                print(res)
+                if res[1] == False:
+                    self._write_advise(res)
+                    self._open_merge_request(branch_name, ['bot'], ["Pipfile.lock"])
```

Comment:
Can be label taken from the `.thoth.yaml` config file stored in user's repo so users can configure the behavoir of this bot? See this default configuration file - https://github.com/thoth-station/thamos/blob/945c095a60b7562c93cda3bd1bcf67130b5c0d11/thamos/data/defaultThoth.yaml#L36-L51

Reply:
Yeah, you're right.  I'll either link to the `thamos` repository to give an example or copy the example from there into the README.  This gets rid of the need to add other options to the wrapper because `advise()` already uses `.thoth.yaml` in the base repository when creating advise request